CONTRIBUTING.md

```diff
@@ -0,0 +1,159 @@
+    ```
+
+6. To ensure that your development environment is set up correctly, run the
+test suite. If the tests pass, you're ready to start developing on Cladetime!
+
+    ```bash
+    uv run pytest
+    ```
+
+> [!TIP]
+> Prefixing python commands with `uv run` instructs uv to run the command
+> in the project's virtual environment, even if it isn't activated.
+
+### Making and submitting changes
+
+To get started, create a new branch for your changes. The branch name should
+use the format
+`<your initials>/<change-name>/<issue-number-if-applicable>`.
+
+After you've created a new branch, make and test your changes.
+We recommend making incremental changes and small commits so it's
+easier for code reviewers to understand the updates.
+
+Cladetime uses the following guidelines and tools to maintain code quality.
+If you want to contribute but have questions about these tools, free to ask
+in your issue or pull request (PR). We're happy to help!
+
+- [Ruff](https://docs.astral.sh/ruff/), to lint and format code. GitHub will
+automatically run Ruff on your pull request and fail if there are any issues.
+You can run Ruff locally (and fix errors) with `uv run ruff --fix`.
+- [pytest](https://docs.pytest.org/en/stable/), to run tests. Large changes
+and new features should be accompanied by new tests.
+- Python module, class, and function docstrings follow the
+[NumPy Style Guide](https://numpydoc.readthedocs.io/en/latest/format.html).
+[Some good examples of NumPy-style docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).
+- [Read the Docs](https://readthedocs.org/), to build and host the project's
+[documentation](https://cladetime.readthedocs.io/en/latest/).
+If you're adding or updating features that impact the documentation, please
+include those updates with your changes (see below for more information about
+updating Cladetime's documentation).
+
+When your changes are complete and you're ready to submit a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests),
+follow these steps:
+
+1. If the repository has a CHANGELOG.md file, update it with a description of
+your changes.
+2. Push the changes to your forked repository.
+3. From your forked repository on GitHub, create a pull request for your branch, targeting
+the `main` branch of the original `cladetime` repository.
+
+4. A team member will review the pull request and be in touch.
+
+### Adding new dependencies
+
+If your change requires a new dependency in the project, add it as follows:
+
+1. Add the dependency to the `dependencies` section of [`pyproject.toml`](pyproject.toml).
+2. Update the requirements files:
+
+    ```bash
+    uv pip compile pyproject.toml -o requirements/requirements.txt && uv pip compile pyproject.toml --extra dev -o requirements/requirements-dev.txt
+    ```
```

Comment:
should this include stuff for how to build `requirements/requirements-doc.txt`?

Reply:
Good question--I was operating under the assumption that the docs requirements will be pretty stable and contributors won't have to worry about them (I know, I know)

Reply:
@elray1 I added that info to the documentation section